packages/devtools_app/lib/src/timeline/flutter/timeline_flame_chart.dart

```diff
@@ -222,25 +76,85 @@ class FullTimelineFlameChart
   static int asyncGuidelineOffset = 1;
 
   @override
-  _FullTimelineFlameChartState createState() => _FullTimelineFlameChartState();
+  _TimelineFlameChartState createState() => _TimelineFlameChartState();
 }
 
-class _FullTimelineFlameChartState
-    extends FlameChartState<FullTimelineFlameChart, TimelineEvent> {
+class _TimelineFlameChartState
+    extends FlameChartState<_TimelineFlameChart, TimelineEvent> {
   /// Stores the [FlameChartNode] for each [TimelineEvent] in the chart.
   ///
   /// We need to be able to look up a [FlameChartNode] based on its
   /// corresponding [TimelineEvent] when we traverse the event tree.
-  final Map<TimelineEvent, FlameChartNode> chartNodesByEvent = {};
+  final chartNodesByEvent = <TimelineEvent, FlameChartNode>{};
 
   /// Async guideline segments drawn in the direction of the x-axis.
-  final List<HorizontalLineSegment> horizontalGuidelines = [];
+  final horizontalGuidelines = <HorizontalLineSegment>[];
 
   /// Async guideline segments drawn in the direction of the y-axis.
-  final List<VerticalLineSegment> verticalGuidelines = [];
+  final verticalGuidelines = <VerticalLineSegment>[];
+
+  final eventGroupStartXValues = Expando<double>();
 
   int widestRow = -1;
 
+  TimelineController _timelineController;
+
+  TimelineFrame _selectedFrame;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    final newController = Controllers.of(context).timeline;
+    if (newController == _timelineController) return;
+    _timelineController = newController;
+
+    addAutoDisposeListener(
+      _timelineController.selectedFrameNotifier,
+      _handleSelectedFrame,
+    );
+  }
+
+  void _handleSelectedFrame() async {
+    final TimelineFrame selectedFrame =
+        _timelineController.selectedFrameNotifier.value;
+    if (selectedFrame != null) {
+      setState(() {
+        _selectedFrame = selectedFrame;
+      });
+
+      // Vertically scroll to the UI event group.
+      final verticalScrollOffset =
+          eventGroupStartXValues[widget.data.eventGroups[TimelineData.uiKey]];
+      await verticalScrollController.animateTo(
+        verticalScrollOffset,
+        duration: shortDuration,
+        curve: defaultCurve,
+      );
+
+      // Zoom the frame into view.
```

Comment:
we need to make sure the animation still makes sense when we get here otherwise things will get janky if a couple animations trigger in succession. same for all other animations within this method.
An easy way to track it would be to make sure _selectedFrame == selectedFrame and bail out otherwise.

Reply:
Like so? (ptal)

Reply:
yep!